applications/vessel-history/src/features/search/search.slice.ts

```diff
@@ -1,56 +1,94 @@
-import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { createSelector, createSlice } from '@reduxjs/toolkit'
 import { VesselSearch } from '@globalfishingwatch/api-types'
 import { RootState } from 'store'
+import { AsyncReducerStatus } from 'utils/async-slice'
+import { selectQueryParam } from 'routes/routes.selectors'
+import { fetchVesselSearchThunk } from './search.thunk'
 
-export type SearchSlice = {
-  query: string
+export type CachedVesselSearch = {
   offset: number
   searching: boolean
   total: number | null
-  vessels: VesselSearch[]
+  vessels: VesselSearch[] | null
+}
+
+const searchInitialState = {
+  vessels: null,
+  total: 0,
+}
+
+export type CachedQuerySearch = {
+  [key: string]: CachedVesselSearch
+}
+
+export type SearchSlice = {
+  status: AsyncReducerStatus
+  queries: CachedQuerySearch
```

Comment:
Really like this 👏

Reply:
this is realted to my comment of the fetch